File: talks/tpcig2018/examples/oop/sub-override.pl
#!/usr/bin/env perl
use v5.12;
use Sub::Override;

sub foo { say 'original sub ' }
foo();
{
    my $token = Sub::Override->new(
        foo => sub { say 'overridden sub' },
    );
    foo();    # overridden sub
}

# $token goes out of scope...
foo();        # original sub

File: talks/tpcig2018/examples/oop/sub-override.lua
local SubOverride = require 'Sub.Override'

function foo()
  print('original sub')
end
foo() -- original sub

do
  local token = SubOverride:new({
    foo = function()
      print('overridden sub')
    end
  })
  foo() -- overridden sub
end

foo() -- overridden sub (WTH?!)
